CRASM-3416: Fix KEV and Ransomware columns empty on CSV export

jsalinasnttdata opened the pull request:
<!-- GitHub renders PRs such that soft line breaks are treated as hard
line breaks.  In order to make this template render as expected we
therefore have to avoid soft breaks and therefore will offend
markdownlint with long lines.  This is the reason for the markdownline
disable directive just below.

For more details see:
https://github.com/github/markup/issues/1050#issuecomment-294654762 -->
<!-- markdownlint-disable MD013 -->
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

Fixed KEV and Ransomware columns appearing empty in CSV exports from the Vulnerabilities table. The issue was that MUI DataGrid exports raw data values rather than formatted display values, so null/undefined boolean values appeared as empty cells in the CSV.

The solution converts KEV and Ransomware values to display strings ("Yes", "No", "N/A") at the data mapping layer, ensuring both UI display and CSV export show consistent values.

## 💭 Motivation and context ##

<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

Resolves CRASM-3416: Users reported that KEV and Ransomware columns were empty in CSV exports despite showing proper "Yes"/"No" values in the UI table.

Root cause: MUI DataGrid's `GridToolbarExport` exports the raw row data, not the processed values from `renderCell` or `valueFormatter`. When KEV/Ransomware fields contained null/undefined values, they exported as empty cells.

Solution: Pre-format the data as strings in the `vulRows` mapping and update column definitions to work with string values instead of booleans.

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

- Verified KEV and Ransomware columns display correctly in the UI table
- Confirmed CSV export now includes "Yes", "No", or "N/A" values for KEV and Ransomware columns
- Tested filtering functionality still works with the new string-based values
- Validated that filter API calls correctly convert display strings back to boolean/null values

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.
- [ ] Bump major, minor, patch, pre-release, and/or build versions [as appropriate](https://semver.org/#semantic-versioning-specification-semver) via the `bump_version` script *if* this repository is versioned *and* the changes in this PR [warrant a version bump](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api).
- [ ] Create a pre-release (necessary if and only if the pre-release version was bumped).

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been -->
<!-- approved. -->

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [ ] Create a release (necessary if and only if the version was bumped).

jsalinasnttdata committed:
refs CRASM-3416: Fix KEV and Ransomware columns empty on CSV export

- Convert KEV and Ransomware values from boolean/null to display strings in vulRows mapping
- Update DataGrid column definitions to use string valueOptions ('Yes', 'No', 'N/A')
- Remove complex valueGetter logic that was causing null values in exports
- Simplify renderCell functions to display pre-formatted string values

jsalinasnttdata committed:
Merge branch 'develop' into jsalinas-vs-fix-vuln-export-kev-ransomware-columns-CRASM-3416

jsalinasnttdata committed:
Merge branch 'develop' into jsalinas-vs-fix-vuln-export-kev-ransomware-columns-CRASM-3416

jsalinasnttdata committed:
refs CRASM-3416: Pre-commit hook

jsalinasnttdata committed:
refs CRASM-3416: Extract KEV/Ransomware conditional logic from return object

jsalinasnttdata committed:
refs CRASM-3416: Add unit tests for KEV and Ransomware column formatting

- Add comprehensive test coverage for vulnerability export functionality
- Test boolean to string transformation (true/false/null -> Yes/No/N/A)
- Verify mixed KEV and ransomware status combinations are displayed correctly
- Update Vuln type definition to allow null values for is_kev and is_kev_ransomware fields
- Ensure CSV export formatting works as expected for all value types

jsalinasnttdata committed:
refs CRASM-3416: Pre-commit hook

jsalinasnttdata committed:
Merge branch 'develop' into jsalinas-vs-fix-vuln-export-kev-ransomware-columns-CRASM-3416

jsalinasnttdata committed:
refs CRASM-3416: KEV/Ransomware filtering tests

- Test server-side filtering with actual payload format: {"filters": {"is_kev_ransomware": true}}

rapidray12 reviewed (commented):
valueOptions: [
  { value: true, label: 'Yes' },
  { value: false, label: 'No' }
  { value: 'Yes', label: 'Yes' },
  { value: 'No', label: 'No' },
  { value: 'N/A', label: 'N/A' }
],

The missing comma after { value: false, label: 'No' }  is most likely  a syntax error. Tt will result in unkown behaviour

rapidray12 commented on frontend/src/pages/Vulnerabilities/Vulnerabilities.tsx:
```diff
@@ -422,33 +429,18 @@ export const Vulnerabilities: React.FC<VulnerabilitiesProps> = ({
-          { value: true, label: 'Yes' },
-          { value: false, label: 'No' }
+          { value: 'Yes', label: 'Yes' },
+          { value: 'No', label: 'No' },
```
Missing comma after { value: false, label: 'No' }  is most likely a syntax error.

jsalinasnttdata replied:
@rapidray12 I believe you may be looking at an outdated diff or mixed old/new code view. The current implementation actually uses only string values in the valueOptions:

```
valueOptions: [
  { value: 'Yes', label: 'Yes' },
  { value: 'No', label: 'No' },
  { value: 'N/A', label: 'N/A' }
]
```

Please let me know if I missing something here.

rapidray12 commented on frontend/src/pages/Vulnerabilities/Vulnerabilities.tsx:
```diff
@@ -464,30 +456,15 @@ export const Vulnerabilities: React.FC<VulnerabilitiesProps> = ({
-          { value: false, label: 'No' }
+          { value: 'Yes', label: 'Yes' },
+          { value: 'No', label: 'No' },
+          { value: 'N/A', label: 'N/A' }
```
Same issue as noted earlier, you probably want a comma after  value: false, label: 'No'

jsalinasnttdata replied:
@rapidray12 same as comment above.

rapidray12 commented on frontend/src/tests/pages/Vulnerabilities/vulnerabilitiesTable.test.tsx:
```diff
@@ -270,6 +270,296 @@ describe('Vulnerabilities component', () => {
+    expect(noElements.length).toBeGreaterThan(0); // Should show "No", not false
+
+    // Test server-side filter with boolean true (matches real payload structure)
+    const mockTrueResponse = await apiPostMock('/api/vulnerabilities', {
```
This test bypasses the DataGrid and calls apiPostMock directly, it only proves the mock test is working correctly

jsalinasnttdata replied:
@rapidray12 I see what you mean! I am fixing this now.

jsalinasnttdata replied:
@rapidray12 I've replaced that test with a proper component integration test that:

1. Actually renders the <Vulnerabilities /> component
2. Validates the data transformation from server boolean values to display strings
3. Verifies that "Yes"/"No"/"N/A" appear in the UI instead of raw boolean values
4. Tests the complete data flow through the component

The new test validates the actual fix rather than just mock behavior. All 11 tests still pass.

rapidray12 commented on frontend/src/pages/Vulnerabilities/Vulnerabilities.tsx:
```diff
@@ -319,15 +319,22 @@ export const Vulnerabilities: React.FC<VulnerabilitiesProps> = ({
+        const kevStatus =
+          vuln.is_kev === null ? 'N/A' : vuln.is_kev ? 'Yes' : 'No';
+
+        const ransomwareStatus =
```
I got this comment from ChatGPT. Treat it accodingly :) 
Bug: is_kev and is_kev_ransomware are defined twice in the row object; the string version overwrites the boolean-normalized version. Please separate boolean and display fields (e.g., is_kev vs is_kev_display) and adjust column configs accordingly.

jsalinasnttdata replied:
@rapidray12 I do see the concern and you're absolutely right, separating the data layer from the presentation layer and avoiding mutation is a way better practice. I'll make the update.

jsalinasnttdata committed:
refs CRASM-3416: Separate data and display layers for KEV/Ransomware

- Preserve original boolean values from server (is_kev, is_kev_ransomware)
- Add string display fields for UI consistency (is_kev_display, is_kev_ransomware_display)
- Update DataGrid columns to use display fields

jsalinasnttdata committed:
refs CRASM-3416: Fix bypassed DataGrid test to validate actual component behavior

ameliav reviewed (approved):
Tested and LGTM!
